src/parser/whamm_parser.rs

```diff
@@ -217,6 +216,214 @@ pub fn process_pair(whamm: &mut Whamm, script_count: usize, pair: Pair<Rule>, er
+                                        vec![
+                                            Rule::TY_I32,
+                                            Rule::TY_BOOL,
+                                            Rule::TY_STRING,
+                                            Rule::TY_TUPLE,
+                                            Rule::TY_MAP,
+                                            Rule::ID,
+                                        ],
+                                        vec![rule],
+                                    );
+                                    // should have exited above (since it's a fatal error)
+                                    unreachable!();
+                                }
+                            }
+                        }
+                        let param_id_local = Expr::VarId {
+                            is_comp_provided: false,
+                            name: arg_name,
+                            loc: None,
+                        };
+                        //arg holds Vec<(VarId, DataType)>
+                        args.push((param_id_local, type_local));
+                    }
+                    Rule::block => {
+                        let body_result = block_from_rule(p, err);
+                        body = match body_result {
+                            Ok(body) => body,
+                            Err(errors) => {
+                                err.add_errors(errors);
+                                Block {
+                                    stmts: vec![Statement::dummy()],
+                                    loc: Some(Location::from(
+                                        &fn_name_line_col,
+                                        &fn_name_line_col,
+                                        None,
+                                    )),
+                                }
+                            }
+                        };
+                    }
+                    rule => {
+                        err.parse_error(
+                            true,
+                            Some(
+                                "Error in proceeing function datatype in parameters of FnDef"
+                                    .to_string(),
+                            ),
+                            Some(LineColLocation::from(p.as_span())),
+                            vec![Rule::block, Rule::param, Rule::TYPE],
+                            vec![rule],
+                        );
+                        // should have exited above (since it's a fatal error)
+                        unreachable!();
+                    }
+                }
+            }
+            //convert fn_name_rule from Pair<_, Rule> to FnId
+            let fn_id = FnId {
+                name: fn_name_rule.as_str().parse().unwrap(),
+                loc: Some(Location::from(&fn_name_line_col, &fn_name_line_col, None)),
```

Comment:
I'd think this should be something similar to what's done for `fn_target_line_col` in `type_from_rule`